sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -435,24 +442,38 @@ private[lf] object SBuiltinFun {
   // Text functions
   //
   final case object SBExplodeText extends SBuiltinPure(1) {
-    override private[speedy] def executePure(args: ArraySeq[SValue]): SList =
-      SList(FrontStack.from(Utf8.explode(getSText(args, 0)).map(SText)))
+    override private[speedy] def executePure(args: ArraySeq[SValue], machine: Machine[_]): SList = {
+      val arg0 = getSText(args, 0)
+
+      machine.updateGasBudget(_.BExplodeText.cost(arg0))
+
+      SList(FrontStack.from(Utf8.explode(arg0).map(SText)))
+    }
   }
 
   final case object SBImplodeText extends SBuiltinPure(1) {
-    override private[speedy] def executePure(args: ArraySeq[SValue]): SText = {
+    override private[speedy] def executePure(args: ArraySeq[SValue], machine: Machine[_]): SText = {
       val xs = getSList(args, 0)
       val ts = xs.map {
         case SText(t) => t
         case v => crash(s"type mismatch implodeText: expected SText, got $v")
       }
+
+      machine.updateGasBudget(_.BImplodeText.c)
```

Comment:
Is the `.c` a typo, or is this meant to be a constant? Seems strange that we expose/bind that we have a constant cost here, rather than keeping that concern within the cost model.

Reply:
@paulbrauner-da suggests to rename to cost.